contracts/EscrowAndStaking/CrossChainCollector.sol

```diff
@@ -0,0 +1,408 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+
+import {IRouterClient} from "@chainlink/contracts-ccip/src/v0.8/ccip/interfaces/IRouterClient.sol";
+import {Client} from "@chainlink/contracts-ccip/src/v0.8/ccip/libraries/Client.sol";
+// import {LinkTokenInterface} from "@chainlink/contracts/src/v0.8/shared/interfaces/LinkTokenInterface.sol";
+// import {CCIPReceiver} from "@chainlink/contracts-ccip/src/v0.8/ccip/applications/CCIPReceiver.sol";
+
+import "@openzeppelin/contracts-upgradeable/proxy/utils/Initializable.sol";
+
+// internal
+import "../utils/proxy/solidity-0.8.0/ProxyReentrancyGuard.sol";
+import "../utils/proxy/solidity-0.8.0/ProxyOwned.sol";
+import "../utils/proxy/solidity-0.8.0/ProxyPausable.sol";
+
+import "../interfaces/IStakingThales.sol";
+
+import "./CCIPReceiverProxy.sol";
+
+/// @title - Cross Chain Collector contract for Thales staking rewards
+contract CrossChainCollector is Initializable, ProxyOwned, ProxyPausable, ProxyReentrancyGuard, CCIPReceiverProxy {
+    // Custom errors to provide more descriptive revert messages.
+    error NotEnoughBalance(uint256 currentBalance, uint256 calculatedFees); // Used to make sure contract has enough balance.
+    uint private constant MAX_GAS = 10 * 1e6;
+    IRouterClient private s_router;
+    address public stakingThales;
+
+    address public masterCollector;
+    uint64 public masterCollectorChain;
+
+    mapping(uint => uint64) public chainSelector;
+    mapping(uint64 => uint) public chainSelectorIndex;
+    mapping(uint => address) public collectorAddress;
```

Comment:
what exactly is this variable? Please rename to smth more descriptive. Is it the address of CCIP collector per chain?